### PR TITLE
Improve release script

### PR DESCRIPTION
**Pull request by plbstl**
This PR updates the release script to be slightly more robust, and updates related documentation.

The new implementation:

- Improves debug messages
- Retrieves semver tag from a commit with duplicate tags (e.g. v2, v2.1.2)
- Retrieves the latest release tag from current branch only
- Pushes the new release tag along with necessary commits to remote
- Works with only annotated tags
- Updates the separate major tag to point to the new release
- Terminates the script when a command fails
- Boldens the colored text in terminal
- Uses full flags for commands, where applicable
- Auto-detects major releases
- Creates a `releases/v#` branch for the previous major version, when creating a major release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -217,13 +217,19 @@ GitHub Actions allows users to select a specific version of the action to use,
...
+   local data available in your repository.
+1. **Prompting for a new release tag:** The user is then prompted to enter a
+   new release tag. To assist with this, the script displays the tag retrieved
+   in the previous step, and validates the format of the inputted tag (vX.X.X).
+   The user is also reminded to update the version field in package.json.
+1. **Tagging the new release:** The script then tags a new release and syncs
+   the separate major tag (e.g. v1, v2) with the new release tag
+   (e.g. v1.0.0, v2.1.2). When the user is creating a new major release, the
+   script auto-detects this and creates a `releases/v#` branch for the previous
+   major version.
```
Comment (ncalteen) [inline README.md:231]: What would be the reason for creating a separate `releases/v#` branch? Wouldn't the release tag already cover this?

Comment (plbstl) [reply README.md:231]: I was thinking, to allow patch fixes for a previous release.

Comment (ncalteen) [reply README.md:231]: Sorry for the delay. That makes sense to me!

**Review by ncalteen — APPROVED** [review]

**Comment by ncalteen** [comment]
One small nit, it look like the README is not formatted with prettier. Can you double check that? If you run `npm run format:write` that should take care of it.

**Review by ncalteen — APPROVED** [review]

**Pull request merged**